src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/StartRequestHandler.kt

```diff
@@ -75,35 +78,29 @@ class StartRequestHandler(private val config: FrejaEidAuthenticatorPluginConfig,
             startAuthentication(requestModel, response)
 
     private fun startAuthentication(requestModel: RequestModel, response: Response): Optional<AuthenticationResult> {
-        //config.userPreferencesManager.saveUsername()
-        if (config.userInfoType.equals(UserInfoType.USERNAME)) {
+        if (config.userInfoType == UserInfoType.SSN) {
             config.userPreferencesManager.saveUsername((requestModel.postRequestModel as UsernameModel).username)
         } else {
             config.userPreferencesManager.saveUsername((requestModel.postRequestModel as EmailModel).email)
         }
-//        ("""
-//            1. Start authentication at Freja.
-//            2. Save the resulting authentication transaction ID in the session.
-//            3. Return HTML that:
-//                A. Informs the user to login on their Freja mobile device
-//                B. Has some JavaScript that polls the /wait endpoint using GET (or head)
-//                C. A form that is POSTed by the poller to the /wait endpoint when the poller is informed that auth is done
-//                """)
 
         val postData: Map<String, Any> = createPostData(config.userInfoType, requestModel)
         val responseData: Map<String, Any> = getAuthTransaction(postData)
+        if (responseData["authRef"] != null) {
+            config.sessionManager.put(Attribute.of("authRef", responseData["authRef"].toString()))
+            throw config.exceptionFactory.redirectException(getWaitHandlerUrl(), RedirectStatusCode.MOVED_TEMPORARILY, emptyMap(), false)
+        }
 
         return Optional.empty()
     }
 
     private fun getAuthTransaction(postData: Map<String, Any>): Map<String, Any> {
 
-        val body = HttpRequest.fromString(config.json.toJson(postData), StandardCharsets.UTF_8)
         val httpResponse = getWebServiceClient(config.environment.getHost())
-                .withPath("/1.0/initAuthentication")
+                .withPath("/authentication/1.0/initAuthentication")
                 .request()
-                .contentType("application/json")
-                .body(body)
+                .contentType("text/plain")
+                .body(HttpRequest.fromByteArray(("initAuthRequest=" + Base64.getEncoder().encodeToString(config.json.toJson(postData).toByteArray())).toByteArray()))
```

Comment:
this line is very long. Better to break apart or at least wrap

Reply:
okay

Reply:
fixed